dbms/src/Storages/Page/V3/Blob/BlobStat.h

```diff
@@ -118,6 +119,34 @@ class BlobStats
 
     using BlobStatPtr = std::shared_ptr<BlobStat>;
 
+    class BlobFileIdManager
+    {
+    public:
+        // Used at restart to add old blob file id to manager
+        void addFileId(BlobFileId file_id);
+
+        BlobFileId nextFileId(PageType page_type, const std::lock_guard<std::mutex> &);
+
+        static PageType getBlobFileType(BlobFileId file_id);
+
+    private:
+        static inline bool isRaftFileId(BlobFileId file_id)
+        {
+            return file_id >= FIRST_RAFT_FILE_ID;
+        }
```

Comment:
I would recommend use an extendable structure for `PageType`. For example:

```
enum class PageType : UInt8
{
    Normal = 0;
    PageData = 1;

    // Support at most 10 types
    TypeCountLimit = 10;
}
```

And then 

```
        static inline bool isPageType(PageType t)
        {
            return file_id % PageType::TypeCountLimit == static_cast<UInt8>(t);
        }
```

Reply:
And we advance page_id by `TypeCountLimit`. Also I think `PageType` should not put inside PageStorage module. It should be an outside feature. And we can control the GC logical by configurations based on `PageType`.

Reply:
It means `PageStorage` should support configuration by passed-in `PageTypes`.

Reply:
I don't see what currently makes it extendable and why it is not extendable previously. `PageType` is just an in memory type, so its value doesn't matter. There is nothing to prevent add new types in the future.

Reply:
Fixed.